components/service/glean/docs/metrics/adding-new-metrics.md

```diff
@@ -210,5 +442,140 @@ assertEquals(11, snapshot.sum)
+   `day`
+
+Carefully consider the required resolution for recording your metric, and choose
+the coarsest resolution possible.
 
+The device's offset from UTC is recorded and sent with the datetime value in the ping.
+
+Now that the datetime is defined in `metrics.yaml` you can use the metric to
+record datetimes in the application's code:
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Install
+
+// ...
+Install.firstRun.set() // Records "now"
+Install.firstRun.set(Calendar(2019, 3, 25)) // Records a custom datetime
+```
+
+There are test APIs available too.
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Install
+Glean.enableTestingMode()
+
+// Was anything recorded?
+assertTrue(Install.firstRun.testHasValue())
+// Was it the expected value? 
+assertEquals("2019-03-25-05:00", Install.firstRun.testGetValueAsString())
+```
+
+## UUIDs
+
+UUIDs are used to record values that uniquely identify some entity, such as a client id.
+
+You first need to add an entry for it to the `metrics.yaml` file:
+
+```YAML
+user:
+  client_id:
+    type: uuid
+    description: >
+      A unique identifier for the client's profile
+    ...
+```
+
+Now that the UUID is defined in `metrics.yaml`, you can use the metric to record
+values in the application's code.
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.User
+
+// ...
+User.clientId.generateAndSet() // Generate a new UUID and record it
+User.clientId.set(UUID.randomUUID())  // Set a UUID explicitly
+```
+
+There are test APIs available too.
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.User
+Glean.enableTestingMode()
+
+// Was anything recorded?
+assertTrue(User.clientId.testHasValue())
+// Was it the expected value? 
+assertEquals(uuid, User.clientId.testGetValue())
+```
+
+## Labeled metrics
+
+Some metrics can be used as *labeled* variants. This means that for a single metric entry you define in `metrics.yaml`, you can record into multiple metrics under the same name, each identified by a different string label.
+
+This is useful when you need to break down metrics by a label known at build time or run time. For example:
+- When you want to count a different set of subviews that users interact with, you could use `viewCount["view1"].add()` and `viewCount["view2"].add()`.
+- When you want to count errors that might occur for a feature, you could use `errorCount[errorName].add()`.
+
+**Note**: Be careful with using arbitrary strings as labels and make sure they can't accidentally contain identifying data (like directory paths or user input).
+
+Say you're adding a new counter for errors that can occur when loading a resource from a REST API. First you need to add an entry for the counter to the `metrics.yaml` file:
+
+```YAML
+updater:
+  load_error:
+    type: counter
+    labeled: true # This makes it a labeled counter.
+    labels: # This is optional, if provided it limits the set of labels you can use.
```

Comment:
I would make this stand out and encourage users to use `labels`, unless labels are not known beforehand.